src/Features/CSharp/Portable/UseDefaultExpression/CSharpUseDefaultExpressionCodeFixProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseDefaultExpression
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseDefaultExpression), Shared]
+    internal sealed class CSharpUseDefaultExpressionCodeFixProvider : CodeFixProvider
+    {
+        private const string CS8313 = nameof(CS8313); // A default literal 'default' is not valid as a case constant. Use another literal (e.g. '0' or 'null') as appropriate. If you intended to write the default label, use 'default:' without 'case'.
+        private const string CS8363 = nameof(CS8363); // A default literal 'default' is not valid as a pattern. Use another literal (e.g. '0' or 'null') as appropriate. To match everything, use a discard pattern 'var _'.
+        private const string CS8107 = nameof(CS8107); // Feature is not available in C# 7.0. Please use language version X or greater.
```

Comment:
@jcouv When I get one of these errors of "Feature X is not available", is it possible to determine which feature this is (I need to know this is specifically feature 'default literal'). I couldn't find anything in the diagnostic properties. Do you have an idea? Thanks.

Currently I'm just checking that the token found at the diagnostic location is a default literal, and that's probably sufficient, but it doesn't feel right.

Reply:
yeah, i'd love something cleaner here as well.

Reply:
I'd be tempted to give `default` a dedicated error code in this case (ie. change compiler) as we did for some other "feature is not available" features.

Reply:
@jcouv But that wouldn't work for other features. Why not just add something to the diagnostic properties, similar to RequiredLanguageVersion? The way I imagine it, there would be a property called LanguageFeature with either a value of "default literal" (always the english string for the feature, so that it can be switched on programatically) or something like IDS_FeatureDefaultLiteral (but that would probably require making the compiler enum public).

Reply:
That's possible for sure, but it seems a rather big hammer (especially making enum public) and I'm not yet sure about the generalized problem (how often do we need this?).

Reply:
It would be useful for code fixes similar to this, which see a language feature you cannot use in this language version and offer to convert it to something you can use. For example, a code fix to change out var into a declared variable, or a code fix from async Main into what the compiler would generate.

Reply:
Relates to https://github.com/dotnet/roslyn/issues/19871 (making the language version diagnostic more parameterized and structured)